### PR TITLE
Add larivaar settings and save them in Async Storage

### PR DESCRIPTION
**Pull request by Gauravjeetsingh**
- Save the larivaar assist and current Ang in storage
- Add a toggle switch for larivaar

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -1,73 +1,61 @@
...
-  Button,
-  SafeAreaView,
-  ScrollView,
-  StatusBar,
-  Switch,
-  TextInput,
-  useColorScheme,
-  View,
-} from 'react-native';
+import React, {useEffect, useState} from 'react';
```
Comment (theamanjs) [inline App.tsx:1]: We are not using airbnb eslint style guide. Let's use that veerji

Already in the issues https://github.com/KhalisFoundation/learn-larivaar/issues/31

File: App.tsx [hunk 2]
```diff
@@ -1,73 +1,61 @@
...
+  const saveLarivaar = async (newValue: boolean) => {
+    await setItem(JSON.stringify({enabled: newValue, assist: larivaarAssist}));
+    setLarivaar(newValue);
+  };
+
+  const value = {larivaarAssist, saveLarivaarAssist, larivaar, saveLarivaar};
+
+  useEffect(() => {
+    readItemFromStorage();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (theamanjs) [inline App.tsx:48]: I think we can remove this comment from here and add readItemFromStorage as dependency to fix the issue

File: App.tsx [hunk 3]
```diff
@@ -1,73 +1,61 @@
...
   return (
-    <SafeAreaView>
+    <LarivaarContext.Provider value={value}>
       <NavigationContainer>
-        <Switch
-          value={larivaarAssist}
-          onValueChange={() => setLarivaarAssist(!larivaarAssist)}
-        />
+        <Drawer.Navigator drawerContent={props => getSettings(props)}>
+          <Drawer.Screen name="Learn Larivaar" component={Launchpad} />
```
Comment (theamanjs) [inline App.tsx:55]: i18n needs to be added to the project as well. The more we defer i18n, the harder it becomes to implement. You can do that in upcoming PRs

File: components/Ang/utils/bakePankti.tsx [hunk 4]
```diff
@@ -2,12 +2,19 @@ import React from 'react';
 import {View, Text} from 'react-native';
 import {getStyle} from './getStyle';
 
-export const bake = (verse: string, larivaarAssist: Boolean) => {
+export const bakePankti = (
+  verse: string,
+  larivaar: Boolean,
+  larivaarAssist: Boolean,
+) => {
```
Comment (theamanjs) [inline components/Ang/utils/bakePankti.tsx]: Recommended to have these arguments in object. Here's the benefit:

1. We don't have to firmly abide by the order of arguments while passing in function call
2. Extending the number of arguments without worrying about existing code is easy. Reduces the possibility of breaking change in the future.

File: components/Launchpad/lauchpad.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  };
+
+  const saveCurrentAng = async (newValue: number) => {
+    await setItem(newValue.toString());
+    setInputAng(newValue);
+  };
+
+  useEffect(() => {
+    readItemFromStorage();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (theamanjs) [inline components/Launchpad/lauchpad.tsx:39]: this can be fixed. Similar issue mentioned above

File: styles/layout.tsx [hunk 6]
```diff
@@ -17,4 +17,14 @@ export const layoutStyles = StyleSheet.create({
...
   },
+  settingContainer: {
+    marginTop: 30,
+    padding: 16,
+  },
+  sidebar: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    marginTop: 16,
```
Comment (theamanjs) [inline styles/layout.tsx:28]: Don't we need to specify `display: 'flex'` too?

**Review by theamanjs — CHANGES_REQUESTED** [review]

File: styles/layout.tsx (hunk 6 above)
Comment (Gauravjeetsingh) [reply styles/layout.tsx:28]: display is by default flex here, so not needed.

File: components/Launchpad/lauchpad.tsx (hunk 5 above)
Comment (Gauravjeetsingh) [reply components/Launchpad/lauchpad.tsx:39]: I tried this, and we even had a call about this, fixing the linting issue, makes this useEffect run on every change, we only want it to run once.

File: App.tsx (hunk 1 above)
Comment (Gauravjeetsingh) [reply App.tsx:1]: Will get to this, after all current PRs are merged, and then will add another PR for this.

File: App.tsx (hunk 2 above)
Comment (Gauravjeetsingh) [reply App.tsx:48]: Tried doing that way, but it didn't work. It keeps running the useEffect hook multiple times.

File: App.tsx (hunk 3 above)
Comment (Gauravjeetsingh) [reply App.tsx:55]: Hnji veerji, will do.

Comment (Gauravjeetsingh) [reply App.tsx:55]: Created an issue for this here https://github.com/KhalisFoundation/learn-larivaar/issues/36

File: components/Ang/utils/bakePankti.tsx (hunk 4 above)
Comment (Gauravjeetsingh) [reply components/Ang/utils/bakePankti.tsx]: Done veerji

**Pull request merged**